services/storage/src/simcore_service_storage/datcore_adapter/datcore_adapter.py

```diff
@@ -227,7 +227,8 @@ async def get_file_download_presigned_link(
         dict[str, Any],
         await _request(app, api_key, api_secret, "GET", f"/files/{file_id}"),
     )
-    return cast(AnyUrl, parse_obj_as(AnyUrl, file_download_data["link"]))  # mypy
+    url: AnyUrl = parse_obj_as(AnyUrl, file_download_data["link"])
```

Comment:
Is the `: AnyUrl` annotation needed here? It looks to me like `parse_obj_as` is a `Generic`.

Reply:
@colinRawlings if you have some knowledge about this... mypy does not appreciate the Generic.
pylance is happy, but mypy is not.
Either it is a cast which I would like to, or using the annotation. Do you have other ideas?

Reply:
as a referrence, it seems to be a pydantic issue:
https://github.com/pydantic/pydantic/issues/4556

Reply:
Interesting, I have only experience with `pylance`. I don't think I agree with the `mypy` team's decision to:

> reject the idea of implementing a feature that allows mypy to trust an imported package's type hints as is